kore/src/Kore/Simplify/Predicate.hs

```diff
@@ -180,9 +148,11 @@ simplify sideCondition original =
                 ForallF forallF ->
                     traverse worker (Forall.refreshForall avoid forallF)
                         >>= simplifyForall sideCondition
+                EqualsF equalsF@(Equals _ _ term _) ->
+                    simplifyEquals sideCondition (termLikeSort term)
+                        =<< traverse simplifyTerm equalsF
                 InF inF ->
                     simplifyIn sideCondition =<< traverse simplifyTerm inF
-                _ -> simplifyPredicateTODO sideCondition predicate & MultiOr.observeAllT
       where
         _ :< predicateF = Recursive.project predicate
         ~avoid = freeVariableNames sideCondition
```

Comment:
Aren't `where` bindings lazy by default? What is this `~` for?

This line wasn't changed, but asking anyway while we're here.

Reply:
We have `Strict` on by default, maybe this changes the behavior? I don't know for certain, though.